### PR TITLE
Entity definition data producers.

### PR DESCRIPTION
**Pull request by rthideaway**
One more port with entity definition data producers.

### CODE REVIEW

File: src/Plugin/GraphQL/DataProducer/EntityDefinition/Fields/Status.php [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+    elseif ($entity_definition_field instanceof FieldConfig) {
+      /** @var \Drupal\field\Entity\FieldConfig $entity_definition_field */
+      return $entity_definition_field->status();
+    }
+    elseif ($entity_definition_field instanceof BaseFieldOverride) {
+      /** @var \Drupal\Core\Field\Entity\BaseFieldOverride $entity_definition_field */
+      return $entity_definition_field->status();
+    }
+
+    return NULL;
```
Comment (klausi) [inline src/Plugin/GraphQL/DataProducer/EntityDefinition/Fields/Status.php]: this will throw a fatal error because the function must return a bool

Comment (rthideaway) [reply src/Plugin/GraphQL/DataProducer/EntityDefinition/Fields/Status.php]: good point!

Comment (rthideaway) [reply src/Plugin/GraphQL/DataProducer/EntityDefinition/Fields/Status.php]: wow, somehow forgot about this pr, fixed now. Please re-check

File: src/Plugin/GraphQL/DataProducer/EntityDefinition/EntityDefinition.php [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+ *       required = FALSE
+ *     ),
+ *     "field_types" = @ContextDefinition("FieldTypes",
+ *       label = @Translation("Field types"),
+ *       default = "ALL",
+ *       required = FALSE
+ *     )
+ *   },
+ *   produces = @ContextDefinition("any",
+ *     entity_definition = @Translation("Entity definition")
```
Comment (fubhy) [inline src/Plugin/GraphQL/DataProducer/EntityDefinition/EntityDefinition.php]: The "label" property on the annotation for all the "produces" contexts in this PR are wrong. I assume you copy pasted them and did a string serach and replace all. Should be "label".

**Review by fubhy — CHANGES_REQUESTED** [review]

File: src/Plugin/GraphQL/DataProducer/EntityDefinition/EntityDefinition.php (hunk 2 above)
Comment (rthideaway) [reply src/Plugin/GraphQL/DataProducer/EntityDefinition/EntityDefinition.php]: fixed

**Pull request merged**